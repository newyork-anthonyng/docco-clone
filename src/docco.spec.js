/* eslint-disable no-console, no-global-assign */
/*
 * Mock modules
 */

console = {
  log: jest.fn(),
};
jest.mock('path');
jest.mock('fs-extra');

const fse = require('fs-extra');
const docco = require('./docco');

beforeEach(() => {
  console.log.mockClear();
});

describe('createDocForFile', () => {
  it('should log error when source file was not found', () => {
    fse.pathExists = jest.fn(() => Promise.resolve(false));
    const result = docco.createDocForFile('missingFile.js');

    return result.then(() => {
      expect(console.log).toHaveBeenCalledTimes(1);
      expect(console.log.mock.calls[0]).toMatchSnapshot();
    });
  });

  it('should log error when ', () => {
    fse.pathExists = jest.fn(() => Promise.reject());
    const result = docco.createDocForFile('invalidFile.js');

    return result.then(() => {
      expect(console.log).toHaveBeenCalledTimes(1);
      expect(console.log.mock.calls[0]).toMatchSnapshot();
    });
  });
});

describe('complete', () => {
  it('should copy the correct files to the correct location', () => {
    const result = docco.complete();

    expect(fse.copy).toHaveBeenCalledTimes(1);
    expect(fse.copy.mock.calls).toMatchSnapshot();
    return result;
  });
});

// describe('run');
