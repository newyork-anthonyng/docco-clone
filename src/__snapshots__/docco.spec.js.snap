// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`complete should copy the correct files to the correct location 1`] = `
Array [
  Array [
    "resources/docco.css",
    "docs/docco.css",
  ],
]
`;

exports[`createDocForFile should log error when  1`] = `
Array [
  "[31mðŸ˜“ Uh oh. Something went wrong with invalidFile.js[39m",
]
`;

exports[`createDocForFile should log error when source file was not found 1`] = `
Array [
  "[31mðŸš«ðŸ”Ž Uh oh. We couldn't find the missingFile.js[39m",
]
`;
